external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOutputWriterFactory.scala

```diff
@@ -17,21 +17,24 @@
 
 package org.apache.spark.sql.avro
 
+import org.apache.avro.Schema
 import org.apache.hadoop.mapreduce.TaskAttemptContext
 
 import org.apache.spark.sql.execution.datasources.{OutputWriter, OutputWriterFactory}
 import org.apache.spark.sql.types.StructType
 
 private[avro] class AvroOutputWriterFactory(
     schema: StructType,
-    avroSchema: SerializableSchema) extends OutputWriterFactory {
+    avroJsonSchema: String) extends OutputWriterFactory {
+
+  private lazy val avroSchema = new Schema.Parser().parse(avroJsonSchema)
```

Comment:
the naming is confusing. We have `schema`, `avroJsonSchema` and `avroSchema`. Also add comments about these variables.